gluten-core/src/main/scala/org/apache/gluten/extension/columnar/TransformHintRule.scala

```diff
@@ -502,9 +502,20 @@ case class AddTransformHintRule() extends Rule[SparkPlan] {
             plan.generatorOutput,
             plan.child)
           transformer.doValidate().tagOnFallback(plan)
-        case plan: EvalPythonExec =>
+        case plan: BatchEvalPythonExec =>
           val transformer = EvalPythonExecTransformer(plan.udfs, plan.resultAttrs, plan.child)
           transformer.doValidate().tagOnFallback(plan)
+        case plan: ArrowEvalPythonExec =>
+          if (
+            !BackendsApiManager.getSettings.supportColumnarArrowUdf() ||
+            GlutenConfig.getConf.enableColumnarArrowUDF
+          ) {
```

Comment:
@zhztheplayer For `ArrowEvalPythonExec `, the check can't leverage your refactored check path.

Reply:
No need to add to validator code. Seems to be that we don't do switch-based fallback on both `BatchEvalPythonExec` and `ArrowEvalPythonExec`.